File: check.ts
import * as fs from 'fs';

const text = fs.readFileSync('src/components/Settings.tsx', 'utf8');
const lines = text.split('\n');
let opens = 0;
for (let i = 0; i < lines.length; i++) {
    const line = lines[i].replace(/\/\/.*/, '');
    const numOpen = (line.match(/<div[\s>]/g) || []).length + (line.match(/<motion\.div[\s>]/g) || []).length;
    const numClose = (line.match(/<\/div[\s>]/g) || []).length + (line.match(/<\/motion\.div>/g) || []).length;
    opens += (numOpen - numClose);
    console.log(`${i+1} : ${opens} ( +${numOpen} -${numClose} ) : ${lines[i].trim().substring(0, 40)}`);
}

File: check.py
import re

def check_divs(text):
    opens = 0
    lines = text.split('\n')
    for i, line in enumerate(lines):
        line_clean = re.sub(r'//.*', '', line) # naive remove comments
        # count <div...> and <motion.div...
        # and </div> and </motion.div>
        num_open = len(re.findall(r'<div[\s>]|<motion.div[\s>]', line_clean))
        num_close = len(re.findall(r'</div[\s>]|</motion.div>', line_clean))
        opens += (num_open - num_close)
        print(f"{i+1} : {opens} ( +{num_open} -{num_close} ) : {line.strip()[:40]}")

with open("src/components/Settings.tsx") as f:
    text = f.read()

check_divs(text)
